src/printer.js

```diff
@@ -3880,7 +3880,29 @@ function makeString(rawContent, enclosingQuote) {
     return match;
```

Comment:
> I don't yet completely understand how this works without using lookbehind

It solves the "odd number of backslashes" problem by incrementally consuming pairs of backslashes. A much nicer solution, really.